### PR TITLE
Added Long Term Testing deployment and validation scripts

### PR DESCRIPTION
**Pull request by mbivolan**

### CODE REVIEW

File: Testscripts/Linux/prepare_ltpt_vm.sh [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+function append_date {
+    while read line
+    do
+        echo "[$(date +%Y-%m-%d-%S)] $line"
+    done < "${1:-/dev/stdin}"
+}
+
+function build_ntttcp {
+    if [[ ! $NTTTCP_REPO ]];then
```
Comment (LiliDeng) [inline Testscripts/Linux/prepare_ltpt_vm.sh]: You can use existed function build_ntttcp in utils.sh

File: XML/VMConfigurations/TwoVMs.xml [hunk 2]
```diff
@@ -394,4 +394,90 @@
             </VirtualMachine>
         </ResourceGroup>
     </TwoVM12Disk>
+    <LPST>
+        <isDeployed>NO</isDeployed>
+        <ResourceGroup>
+            <VirtualMachine>
+                <ARMInstanceSize>Standard_DS15_v2</ARMInstanceSize>
+                <RoleName></RoleName>
```
Comment (juhlee-microsoft) [inline XML/VMConfigurations/TwoVMs.xml:402]: RoleName is missing in both VM

File: XML/TestCases/Other.xml [hunk 3]
```diff
@@ -51,4 +51,14 @@
         <Area></Area>
         <Tags>gcov</Tags>
     </test>
+    <test>
+        <testName>LONG-PERF-STRESS-TEST</testName>
```
Comment (juhlee-microsoft) [inline XML/TestCases/Other.xml:55]: why not StressTest.xml?

File: XML/TestCases/Other.xml [hunk 4]
```diff
@@ -51,4 +51,14 @@
...
+        <testScript>DEPLOY-LONG-TERM-VM.ps1</testScript>
+        <setupType>LPST</setupType>
+        <files>.\Testscripts\Linux\utils.sh,.\Testscripts\Linux\prepare_ltpt_vm.sh,.\Testscripts\Linux\CreateRaid.sh</files>
+        <Platform>Azure</Platform>
+        <Category></Category>
```
Comment (juhlee-microsoft) [inline XML/TestCases/Other.xml:60]: Category & Area are missing

File: Utilities/LongTerm/check_values.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import argparse
```
Comment (juhlee-microsoft) [inline Utilities/LongTerm/check_values.py:4]: license info? please check out other files

File: Utilities/LongTerm/parser.ps1 [hunk 6]
```diff
@@ -0,0 +1,283 @@
+param (
```
Comment (juhlee-microsoft) [inline Utilities/LongTerm/parser.ps1:4]: license

File: Utilities/LongTerm/Get-Values.ps1 [hunk 7]
```diff
@@ -0,0 +1,68 @@
+param (
```
Comment (juhlee-microsoft) [inline Utilities/LongTerm/Get-Values.ps1:4]: license missing

File: Utilities/LongTerm/Check-Logs.ps1 [hunk 8]
```diff
@@ -0,0 +1,134 @@
+param (
```
Comment (juhlee-microsoft) [inline Utilities/LongTerm/Check-Logs.ps1:4]: license missing

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

File: XML/VMConfigurations/TwoVMs.xml (hunk 2 above)
Comment (mbivolan) [reply XML/VMConfigurations/TwoVMs.xml:402]: The scripts use the default 'role-0'/'role-1' role names

File: XML/TestCases/Other.xml (hunk 4 above)
Comment (chvalean) [reply XML/TestCases/Other.xml:60]: no need for these, this is a specialized scenario and we don't want it triggered by Category/Area

File: XML/TestCases/Other.xml (hunk 3 above)
Comment (chvalean) [reply XML/TestCases/Other.xml:55]: this is long term testing and not to be executed as part of regular stress tests, so it's fine in the other.xml

File: Testscripts/Linux/prepare_ltpt_vm.sh (hunk 1 above)
Comment (mbivolan) [reply Testscripts/Linux/prepare_ltpt_vm.sh]: removed local function

File: Testscripts/Linux/prepare_ltpt_vm.sh [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+        remote_ip="$SERVER_IP"
+    fi
+    
+    if [[ -d "$LOG_DIR" ]];then
+        rm -rf "$LOG_DIR"
+    fi
+    mkdir $LOG_DIR
+    
+    update_repos
+    install_package make gcc
```
Comment (chvalean) [inline Testscripts/Linux/prepare_ltpt_vm.sh]: consolidate lines 96 & 99

File: Utilities/LongTerm/Check-Logs.ps1 [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+    param (
+        [string] $vmIP,
+        [string] $vmPort,
+        [string] $vmUsername,
+        [string] $vmPassword,
+        [string] $remotePath,
+        [string] $destination
+    )
+
+    echo 'y' | & "C:\Program Files\PuTTY\pscp.exe" -P $vmPort -pw $vmPassword root@${vmIP}:${remotePath} $destination
```
Comment (chvalean) [inline Utilities/LongTerm/Check-Logs.ps1]: pscp directly, without full path.
or you cannot use the download function?

File: Utilities/LongTerm/Check-Logs.ps1 [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+        $result = Download-Log -vmIP $vmData["SERVER_IP"] -vmPort $vmData["SERVER_PORT"] -vmUsername $vmData["VM_USERNAME"] `
+            -vmPassword $vmData["VM_PASSWORD"] -remotePath "$($vmData['REMOTE_LOG_PATH'])/$($SERVER_LOGS[$logType])" `
+            -destination $logName
+        if ($result -eq $true) {
+            Write-Host "Log: $($SERVER_LOGS[$logType]) downloaded successfully"
+        }
+
+        $csvName = "$($logName.Split(".")[0]).csv"
+
+        $failedCollumns = & $PARSER_PATH -LogType $logType -LogPath $logName -CsvDest $csvName -LogDate $Date
```
Comment (chvalean) [inline Utilities/LongTerm/Check-Logs.ps1]: Collumns -> Columns
update all 11 references

File: Utilities/LongTerm/Check-Logs.ps1 (hunk 10 above)
Comment (mbivolan) [reply Utilities/LongTerm/Check-Logs.ps1]: sometimes pscp.exe cannot be found in path, so a full path is needed.
Will replace the path with $env:ProgramFiles so it's safer.
As for the download function, i don't consider it a good idea to import a module only for a function

**Review by chvalean — APPROVED** [review]

**Review by chvalean — APPROVED** [review]

**Comment by chvalean** [comment]
@juhlee-microsoft please approve and merge this, review has been done.

**Review by juhlee-microsoft — APPROVED** [review]

**Pull request merged**